src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -23,11 +23,12 @@ internal class NextLinkOperationImplementation : IOperation
         private readonly OperationFinalStateVia _finalStateVia;
         private readonly RequestMethod _requestMethod;
         private readonly HttpPipeline _pipeline;
+        private readonly ChangeTrackingDictionary<string, string> _additionalOptions;
 
         private string? _lastKnownLocation;
         private string _nextRequestUri;
 
-        public static IOperation Create(HttpPipeline pipeline, RequestMethod requestMethod, Uri startRequestUri, Response response, OperationFinalStateVia finalStateVia)
+        public static IOperation Create(HttpPipeline pipeline, RequestMethod requestMethod, Uri startRequestUri, Response response, OperationFinalStateVia finalStateVia, ChangeTrackingDictionary<string, string>? additionalOptions = default)
```

Comment:
I think what we want to do here is instead replace `RequestMethod requestMethod, Uri startRequestUri` with `Request originalRequest` which will then allow us to pull out the user-agent header in the original request and store in a private field.  Then inside the CreateRequest method if that field is not null set the user-agent to be the same.